File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/map/path/NodeProvider.java
package net.bestia.zoneserver.map.path;

import java.util.Set;

/**
 * Provider for the pathfinder algorithms with nodes from different sources.
 * 
 * @author Thomas Felix
 *
 * @param <T>
 *            The type for the nodes to be supplied.
 */
public interface NodeProvider<T> {

	Set<Node<T>> getConnectedNodes(Node<T> node);

}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/map/path/NodeProvider.kt
package net.bestia.zoneserver.map.path

/**
 * Provider for the pathfinder algorithms with nodes from different sources.
 *
 * @param <T> The type for the nodes to be supplied.
 * @author Thomas Felix
</T> */
interface NodeProvider<T> {

  /**
   * Returns all reachable nodes from the given node. The returned nodes must
   * be provided with a associated walk cost.
   *
   * @param node The node to get the neighbouring and reachable nodes from.
   * @return A set of all reachable nodes with walk cost set.
   */
  fun getConnectedNodes(node: Node<T>): Set<Node<T>>
}
